cpp/src/column/column_view.cpp

```diff
@@ -76,6 +77,67 @@ size_type column_view_base::null_count(size_type begin, size_type end) const
            ? 0
            : cudf::count_unset_bits(null_mask(), offset() + begin, offset() + end);
 }
+
+// Alternative fast hash functions
+// simple prime number multiplication algorithm.
+// Adapted from http://myeyesareblind.com/2017/02/06/Combine-hash-values/#apachecommons
+// constexpr std::size_t combine_hash(std::size_t h1, std::size_t h2) { return h1 * 127 + h2; }
+
+// Struct to use custom hash combine and fold expression
+struct HashValue {
+  std::size_t hash;
+  explicit HashValue(std::size_t h) : hash{h} {}
+  HashValue operator^(HashValue const& other) const
+  {
+    return HashValue{hash_combine(hash, other.hash)};
+  }
+};
+
+template <typename... Ts>
+constexpr auto hash(Ts&&... ts)
+{
+  return (... ^ HashValue(std::hash<Ts>{}(ts))).hash;
+}
+
+struct shallow_hash_impl {
```

Comment:
Does this need to be a function object instead of just a normal function?

Reply:
updated. (it was due to old remnant code)